Kudu.Core/Jobs/AggregrateJobsManagerBase.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Kudu.Contracts.Jobs;
+using Kudu.Contracts.Settings;
+using Kudu.Core.Tracing;
+
+namespace Kudu.Core.Jobs
+{
+    public abstract class AggregrateJobsManagerBase<TJob> where TJob : JobBase, new()
+    {
+        protected JobsManagerBase<TJob> PrimaryJobManager { get; private set; }
+        protected JobsManagerBase<TJob> SecondaryJobManager { get; private set; }
+        protected JobsManagerBase<TJob> WriteJobManager { get; private set; }
+
+        private readonly IDeploymentSettingsManager _settings;
+
+        protected AggregrateJobsManagerBase(Func<JobsManagerBase<TJob>> primaryManagerAction, Func<IEnumerable<string>, JobsManagerBase<TJob>> secondaryManagerAction, IDeploymentSettingsManager settings)
+        {
+            PrimaryJobManager = primaryManagerAction();
+            SecondaryJobManager = secondaryManagerAction(PrimaryJobManager.ListJobs(false).Select(j => j.Name));
```

Comment:
Confusing line could use a comment. e.g.

    // pass the list of primary job names so the second manager can excluded them

Also, use `forceRefreshCache: false` instead of just `false`.

Reply:
done